src/watchmaker/workers/salt.py

```diff
@@ -262,10 +273,47 @@ def _build_salt_formula(self, extract_dir):
                 salt_content_filename
             ))
             self.retrieve_file(self.salt_content, salt_content_file)
+            temp_extract_dir = extract_dir
+            if self.salt_content_path:
+                temp_extract_dir = os.sep.join((
+                    self.working_dir,
+                    'salt-archive'
+                ))
             self.extract_contents(
                 filepath=salt_content_file,
-                to_directory=extract_dir
+                to_directory=temp_extract_dir
             )
+            if self.salt_content_path:
+                self.log.debug(
+                    'Using salt content path: %s',
+                    self.salt_content_path
+                )
+                salt_content_src = os.sep.join((
+                    temp_extract_dir, self.salt_content_path))
+                salt_files_dir = None
+                for (i, path) in enumerate(glob.glob(salt_content_src)):
```

Comment:
any particular rationale for the logic here? seems like it could be a bit simpler...

```
salt_content_glob = glob.glob(salt_content_src)
if length(salt_content_glob) > 0:
    # ... raise
salt_files_dir = salt_content_glob[0]
```

Reply:
yeah, the added logic is there to only raise error when the glob matches multiple directories, not files necessarily. Too much of an edge case, you think?

Reply:
i would say yes, too much an edge case. user can just use a more specific glob, right?